cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceSinkContext.java

```diff
@@ -53,10 +56,11 @@ public void addOutput(final String datasetName) {
 
   @Override
   public void addOutput(final String datasetName, final Map<String, String> arguments) {
+    final Output actualOutput = getOutput(Output.ofDataset(datasetName, arguments));
```

Comment:
getOutput simply returns the datasetName and NullOutputFormatProvider, what happens to the arguments or are they not needed?